java/core/src/main/java/com/microsoft/bond/Unmarshal.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.helpers.ArgumentHelper;
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.FastBinaryReader;
+import com.microsoft.bond.protocol.SimpleBinaryReader;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Resposible for unmarshaling of objects.
+ */
+public final class Unmarshal {
+
+    /**
+     * Unmarshal payload into a Bonded instance.
+     *
+     * @param inputStream input stream
+     * @param bondType    bond type of unmarshalled struct
+     * @param <TStruct>   the type of unmarshalled struct
+     * @return a bonded instance
+     * @throws IOException if an I/O error occurred
+     */
+    public static <TStruct extends BondSerializable> Bonded<TStruct> unmarshal(
+            InputStream inputStream, StructBondType<TStruct> bondType) throws IOException {
+        ArgumentHelper.ensureNotNull(inputStream, "inputStream");
```

Comment:
No it's not. The check for null is made in fromProtocolReader, but I can add it here as well.

---
In reply to: [134010489](https://github.com/Microsoft/bond/pull/571#discussion_r134010489) [](ancestors = 134010489)